### ISSUE TITLE
Figura path changed

### ISSUE THREAD

**Issue by Octelly** [sisby-folk/switchy#50]
Figura is ``org.figuramc.figura`` now as seen [here](https://github.com/FiguraMC/Figura/blob/f2d670ad7e7de21fc69dd3d9b97cca5bb669b382/common/src/main/java/org/figuramc/figura/FiguraMod.java#L1). Also lots of love for this mod <3

**Comment by sisby-folk** [comment]
sure thing - can you give me a minimum version that this change is / will be released in?

**Comment by Octelly** [comment]
There was some drama in the Figura community, so this should be in all current and future versions... I cannot vouch for this however as I'm not a part of the dev team.

The only "current version" Figura I have, where this didn't get changed yet is the one from Modrinth... although that is a temporary build: 
![image](https://github.com/sisby-folk/switchy/assets/29976879/8ef2b568-8fd6-46b7-9388-0c1a910b846f)

All build artifacts (currently only official source of Figura for older MC versions) I've tried have already been updated however...
https://github.com/FiguraMC/Figura/actions

**Comment by sisby-folk** [comment]
🤭 I don't need a life story, I just need a value to put in the quilt.mod.json to prevent loading too-old versions of figura with this update.

We need something with a clear dividing line, where the old package path had (semver) and the new package path has (different semver) - if a new numbered release hasn't come out yet, we'll wait until it does.

if you're unsure, someone else can probably come in and catch us up when apt.

**Comment by Octelly** [comment]
Yeah, that's fair. Just kind of impatient and my attempts at making a custom build have been kind of all over the place :)

**Comment by emifloofy** [comment]
Just looked through the repo, the version that is still in the gradle settings seems to be 0.1.1 ([as seen here](https://github.com/FiguraMC/Figura/blob/a7244489b37314292bce04bcb4b0611772aa3b31/gradle.properties#L12)), however it seems like the next version with these changes is gonna be 0.1.2 ([as seen here](https://github.com/FiguraMC/Figura/issues/49#issuecomment-1662274411))

**Comment by sisby-folk** [comment]
sounds good - when 0.1.2 releases, we'll run a up a new version.

Out of curiosity, are you actually using the Switchy figura lua api? if not, you should be able to remove the entrypoint from the quilt.mod.json by just opening the current Switchy jar in 7zip (should be in switchy-compat or switchy-compat-ui)

**Comment by Octelly** [comment]
Yeah, I am. I do have a fallback if Switchy isn't available, but I would prefer being able to use it.

I got this far with my attempt at compiling a custom version out of desperation, but I appear to be stuck now, so I guess we'll have to wait.

![image](https://github.com/sisby-folk/switchy/assets/29976879/8f5ee67e-6e18-462b-9d10-0283fa5ff659)

**Comment by sisby-folk** [comment]
hmm, did you have the Switchy figura api working before? or is this a new attempt - we haven't heard of anyone having immense success just yet.

**Comment by Octelly** [comment]
Yeah, it is unforunately kind of janky atm (I believe you chatted about this with Aproxia in DMs), but we got it working well enough.

https://github.com/Octelly/octelly-figura/blob/main/src/modules/switchy.yue

EDIT: Aproxia made an issue about it here: #51

**Comment by sisby-folk** [comment]
can you link us through the new developer docs/wiki for figura? we remember them being threadbare but I'll see what I can do on both issues

**Comment by Octelly** [comment]
I wasn't aware there even was a dev wiki...

**Comment by Octelly** [comment]
Anyway, it's almost 5 am and I give up. My Java and its ecosystem knowledge isn't good enough for this. This seems like it should work, but I seem to be missing something. I don't know, guess I'll have to wait for Figura to get their stuff together.

https://github.com/Octelly/switchy-figura-test

**Comment by Octelly** [comment]
[0.1.2 just released](https://github.com/FiguraMC/Figura/releases/tag/0.1.2)
(I'm sorry for being annoying about this, really excited to be able to use the two together again)

**Comment by sisby-folk** [comment]
fixed in 0c6e04a

**Comment by totallytavi** [comment]
I tried to run Switchy (2.8.4+1.20) with Figura (0.1.2+1.20.1) on 1.20.1. I suspect this change hasn't been replicated (Given the last commit was 3 weeks ago) and my game fails to start entirely. Logs extracted from my 
<details><summary>Relevant Logs</summary>
<p>
[02:34:34] [main/INFO]: Loading Minecraft 1.20.1 with Fabric Loader 0.14.22

[02:34:35] [ForkJoinPool-1-worker-6/WARN]: The mod "switchy-compat-ui" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 36 column 12

[02:34:35] [ForkJoinPool-1-worker-7/WARN]: The mod "switchy-ui" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 36 column 12

[02:34:35] [ForkJoinPool-1-worker-8/WARN]: The mod "switchy-core" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 31 column 12

[02:34:35] [ForkJoinPool-1-worker-5/WARN]: The mod "switchy-client" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 35 column 12

[02:34:35] [ForkJoinPool-1-worker-6/WARN]: The mod "switchy-cardinal-ui" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 35 column 12

[02:34:35] [ForkJoinPool-1-worker-14/WARN]: The mod "switchy-cardinal" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 34 column 12

[02:34:35] [ForkJoinPool-1-worker-4/WARN]: The mod "switchy-compat" contains invalid entries in its mod json:
- Unsupported root entry "modmenu" at line 44 column 12

[02:34:35] [main/INFO]: Loading 70 mods:
	- fabric-api 0.90.0+1.20.1
	   |-- fabric-api-base 0.4.30+7abfd51577
	   |-- fabric-api-lookup-api-v1 1.6.35+4d8536c977
	   |-- fabric-biome-api-v1 13.0.12+215bbe9677
	   |-- fabric-block-api-v1 1.0.10+92a0d36777
	   |-- fabric-block-view-api-v2 1.0.0+92a0d36777
	   |-- fabric-blockrenderlayer-v1 1.1.40+b3afc78b77
	   |-- fabric-client-tags-api-v1 1.1.1+97bb207577
	   |-- fabric-command-api-v1 1.2.33+f71b366f77
	   |-- fabric-command-api-v2 2.2.12+b3afc78b77
	   |-- fabric-commands-v0 0.2.50+df3654b377
	   |-- fabric-containers-v0 0.1.63+df3654b377
	   |-- fabric-content-registries-v0 4.0.10+57aed33f77
	   |-- fabric-convention-tags-v1 1.5.4+a1a980da77
	   |-- fabric-crash-report-info-v1 0.2.18+aeb40ebe77
	   |-- fabric-data-generation-api-v1 12.3.2+d7b148e077
	   |-- fabric-dimensions-v1 2.1.53+8536527b77
	   |-- fabric-entity-events-v1 1.5.22+b3afc78b77
	   |-- fabric-events-interaction-v0 0.6.1+e91849a877
	   |-- fabric-events-lifecycle-v0 0.2.62+df3654b377
	   |-- fabric-game-rule-api-v1 1.0.38+b04edc7a77
	   |-- fabric-item-api-v1 2.1.27+b3afc78b77
	   |-- fabric-item-group-api-v1 4.0.11+d7b148e077
	   |-- fabric-key-binding-api-v1 1.0.36+fb8d95da77
	   |-- fabric-keybindings-v0 0.2.34+df3654b377
	   |-- fabric-lifecycle-events-v1 2.2.21+b3afc78b77
	   |-- fabric-loot-api-v2 1.2.0+96dfa95977
	   |-- fabric-loot-tables-v1 1.1.44+9e7660c677
	   |-- fabric-message-api-v1 5.1.8+d7b148e077
	   |-- fabric-mining-level-api-v1 2.1.49+b3afc78b77
	   |-- fabric-model-loading-api-v1 1.0.2+709a987177
	   |-- fabric-models-v0 0.4.1+9386d8a777
	   |-- fabric-networking-api-v1 1.3.10+eeb8eb3677
	   |-- fabric-networking-v0 0.3.50+df3654b377
	   |-- fabric-object-builder-api-v1 11.1.1+6beca84877
	   |-- fabric-particles-v1 1.1.1+201a23a077
	   |-- fabric-recipe-api-v1 1.0.20+b3afc78b77
	   |-- fabric-registry-sync-v0 2.3.2+4df89eb277
	   |-- fabric-renderer-api-v1 3.2.0+39a511ba77
	   |-- fabric-renderer-indigo 1.5.0+39a511ba77
	   |-- fabric-renderer-registries-v1 3.2.45+df3654b377
	   |-- fabric-rendering-data-attachment-v1 0.3.36+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.27+b3afc78b77
	   |-- fabric-rendering-v0 1.1.48+df3654b377
	   |-- fabric-rendering-v1 3.0.7+b3afc78b77
	   |-- fabric-resource-conditions-api-v1 2.3.6+d7b148e077
	   |-- fabric-resource-loader-v0 0.11.9+132c48c177
	   |-- fabric-screen-api-v1 2.0.7+b3afc78b77
	   |-- fabric-screen-handler-api-v1 1.3.29+b3afc78b77
	   |-- fabric-sound-api-v1 1.0.12+b3afc78b77
	   |-- fabric-transfer-api-v1 3.3.1+195226a777
	   \-- fabric-transitive-access-wideners-v1 4.3.0+6c31357e77
	- fabricloader 0.14.22
	- figura 0.1.2+1.20.1
	   |-- com_github_figuramc_luaj_luaj-core 3.0.6-figura
	   |-- com_github_figuramc_luaj_luaj-jse 3.0.6-figura
	   \-- org_java-websocket_java-websocket 1.5.4
	- java 17
	- minecraft 1.20.1
	- switchy 2.8.4+1.20
	   |-- switchy-cardinal 2.8.4+1.20
	   |-- switchy-cardinal-ui 2.8.4+1.20
	   |-- switchy-client 2.8.4+1.20
	   |-- switchy-compat 2.8.4+1.20
	   |-- switchy-compat-ui 2.8.4+1.20
	   |-- switchy-core 2.8.4+1.20
	   |    |-- folk_sisby_kaleido-config 0.1.0+1.1.0-beta.3
	   |    \-- server_translations_api 2.0.0+1.20
	   |         \-- packet_tweaker 0.4.0+1.19.4
	   \-- switchy-ui 2.8.4+1.20

[02:34:35] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/C:/Users/{COMPUTER_USERNAME}/AppData/Roaming/com.modrinth.theseus/meta/libraries/net/fabricmc/sponge-mixin/0.12.5+mixin.0.8.5/sponge-mixin-0.12.5+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT

[02:34:35] [main/INFO]: Compatibility level set to JAVA_16

[02:34:35] [main/INFO]: Compatibility level set to JAVA_17

[02:34:36] [main/WARN]: Error loading class: de/maxhenkel/voicechat/voice/client/RenderEvents (java.lang.ClassNotFoundException: de/maxhenkel/voicechat/voice/client/RenderEvents)

[02:34:36] [main/WARN]: Error loading class: org/figuramc/figura/avatar/Avatar (java.lang.ClassNotFoundException: org/figuramc/figura/avatar/Avatar)

[02:34:36] [main/WARN]: @Mixin target org.figuramc.figura.avatar.Avatar was not found switchy-compat.mixins.json:compat.figura.AvatarMixin from mod switchy-compat

... [I stopped execution, seeing that last message but here's the next message I would've gotten]

[HH:MM:SS] [Render thread/WARN]:
 Injection warning: LVT in net/minecraft/class_757::method_3188(FJLnet/minecraft/class_4587;)V has incompatible changes at opcode 256 in callback fabric-figura-common.mixins.json:render.GameRendererMixin from mod figura->@Inject::renderLevelSaveBobbingOF(FJLnet/minecraft/class_4587;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;ZZLnet/minecraft/class_4184;Lnet/minecraft/class_4587;D)V.
 Expected: [Z, Z, Lnet/minecraft/class_4184;, Lnet/minecraft/class_4587;, D]
    Found: [Z, Lnet/minecraft/class_4184;, Lnet/minecraft/class_4587;, D, F]
Available: [Z, Lnet/minecraft/class_4184;, Lnet/minecraft/class_4587;, D, F, F, I, F, Lnet/minecraft/class_7833;, F]
</p>
</details>

**Comment by sisby-folk** [comment]
I got nothing on that, that class clearly exists on the main branch here, right? <https://github.com/FiguraMC/Figura/blob/1.20/common/src/main/java/org/figuramc/figura/avatar/Avatar.java>

**Comment by totallytavi** [comment]
I'm checking if it's my Java. ~~Would a CurseForge download maybe work?~~ Tried running the Fabric installer. It cannot detect my Java. Working on that.

**Comment by sisby-folk** [comment]
Yeah i can [reproduce this](https://mclo.gs/nC7mrSC), reopening.

**Comment by sisby-folk** [comment]
Looks like it's on figura possibly, i'll open an issue.

**Comment by magnacrabb** [comment]
currently on switchy 2.8.4 (forge 1.20.1) & figura 0.1.3 pre3 (forge), may i ask how to make an edit so that it can refer to the right path?

**Comment by sisby-folk** [comment]
mod has been updated with a PR since this last got messaged, so closing.
